ui/tests/e2e/login.test.ts

```diff
@@ -0,0 +1,56 @@
+import puppeteer from 'puppeteer'
+import { LOGIN_URL, OVERVIEW_URL, PUPPETEER_CONFIG } from './test_config'
+
+describe('Login', () => {
+  let browser
+  beforeAll(async () => {
+    browser = await puppeteer.launch(PUPPETEER_CONFIG)
+  })
+
+  afterAll(() => {
+    browser.close()
+  })
+
+  it(
+    'should login fail by incorrect password',
+    async () => {
+      const page = await browser.newPage()
+      await page.goto(LOGIN_URL)
+
+      const pwdInput = await page.waitForSelector('input#tidb_signin_password')
+      await pwdInput.type('any')
+      await page.click('button#signin_btn')
+
+      const errorContainer = await page.waitForSelector('div.has-error')
```

Comment:
ditto. This test heavily rely on the implementation of ant design itself, which is not good.

Reply:
Tried to add "id" to the component but some places work, some places don't, the "id" can't pass to real dom node, it needs to change the original code much, I think we can do that later.